internal/jwx/jws/sign/sign.go

```diff
@@ -37,7 +37,11 @@ func GetSigningKey(key string, alg jwa.SignatureAlgorithm) (interface{}, error)
 
 		priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
 		if err != nil {
-			return nil, err
+			pkcs8priv, err2 := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err2 != nil {
+				return nil, fmt.Errorf("error parsing private key (%v), (%v)", err, err2)
```

Comment:
Not sure if there's a better way of achieving this - what I want is for both the parse errors to be returned so that the user sees that both PKCS1 and PKCS8 was attempted.

Reply:
🤔 Maybe something like this?
```go
return nil, fmt.Errorf("error parsing private key as PKCS1 (%v) or PKCS8 (%v))
```

But really, can we differentiate the situation based on the first `err`? I hate to bring it up, but it seems like the error string would tell us when to use x509.ParsePKCS8PrivateKey instead: https://golang.org/src/crypto/x509/pkcs1.go?s=1140:1202#:L58

Reply:
You mean we should parse the error string in order to decide if calling `x509.ParsePKCS8PrivateKey` makes sense?

Reply:
Yeah...? I don't like it either :/

Reply:
Hmm yeah the error handling in Go leaves a lot to wish for IMO :/ Not sure if error messages are considered part of the contract or if they could change between releases (however unlikely). I guess we could test for _that_ too, but in this case I'm not sure it would bring us much of value.